### DISCUSSION TITLE
Eager Loading Custom Query or CTE?

### DISCUSSION THREAD

**Discussion by jordantshaw** [sqlalchemy/sqlalchemy#6838] [Usage Questions]
Hi all,

I need to be able to eager load using either selectinload or subqueryload ( this is a one to many relationship) a custom query or CTE.  Normally I would setup the relationship through the table model, however, in this cause the custom query is not actually a table and therefore is not modeled.  Is there a way to setup a relationship for a CTE or query?  Or Better yet can I just specify join conditions dynamically?  Any ideas on how I can get this to work?

Below is a sample of what Im trying to do.

```
# This does not necessarily need to be a CTE, just a custom query of some sort.

SubQuery = (
      select(Table2)
      .outerjoin(Table3)
      .cte(name='subQuery')
  )


query = (
    select(Table2)
    .options(
        selectinload(SubQuery),
    )
    .order_by(order_by))
    .limit(limit)
)

results = session.execute(query).scalars().all()
```

**Comment by CaselIT** [comment]
Hi,

The loader options can only be added for mapped objects.
If the subquery/cte is not dynamic you can map it to an orm model by assigning it to the `__table__` argument. At that point you should be able to use a relationship and use any of the loader documented.

Some examples from the documentation https://docs.sqlalchemy.org/en/14/orm/nonstandard_mappings.html#orm-mapping-arbitrary-subqueries
